### PR TITLE
Implemented #2153 - Upgraded to Servlet 6

### PR DESCRIPTION
**Pull request by dmatej**
Signed-off-by: David Matějček <dmatej@seznam.cz>

### CODE REVIEW

**Comment by dmatej** [comment]
Local build passed including tests

**Comment by pzygielo** [comment]
Could the JDK used in https://ci.eclipse.org/grizzly/job/Grizzly-master_nightly/ be updated, please?

**Comment by dmatej** [comment]
> Could the JDK used in https://ci.eclipse.org/grizzly/job/Grizzly-master_nightly/ be updated, please?

I don't have rights to do that, seems at least switching from JDK8 to JDK11 would be nice ... @arjantijms ?

**Review by arjantijms — APPROVED** [review]

**Comment by arjantijms** [comment]
I'm on it, thanks!

**Comment by arjantijms** [comment]
Update done!

**Comment by pzygielo** [comment]
thanks! and of course - `staging` profile has to be activated - 
https://ci.eclipse.org/grizzly/job/Grizzly-master_nightly/1518/consoleText:
```
Executing Maven:  -B -f /home/jenkins/agent/workspace/Grizzly-master_nightly/pom.xml clean install -Dmaven.test.redirectTestOutputToFile=true -DforkCount=1 -DreuseForks=false -Denforcer.skip=true
...
[ERROR] Failed to execute goal on project grizzly-http-servlet: Could not resolve dependencies for project org.glassfish.grizzly:grizzly-http-servlet:bundle:4.0.0-SNAPSHOT: Could not find artifact jakarta.servlet:jakarta.servlet-api:jar:6.0.0 in eclipse.maven.central.mirror (https://repo.eclipse.org/content/repositories/maven_central/) -> [Help 1]
```

File: modules/http-servlet/src/main/java/org/glassfish/grizzly/servlet/SessionCookieConfig.java [hunk 1]
```diff
@@ -16,19 +17,27 @@
...
 public class SessionCookieConfig implements jakarta.servlet.SessionCookieConfig {
 
-    private String name;
-    private String domain;
-    private String path;
-    private String comment;
-    private boolean httpOnly = true;
-    private boolean secure;
+    private static final int DEFAULT_MAX_AGE = -1;
+    private static final boolean DEFAULT_HTTP_ONLY = false;
```
Comment (hs536) [inline modules/http-servlet/src/main/java/org/glassfish/grizzly/servlet/SessionCookieConfig.java:33]: @dmatej could you tell me the reason for changing this default value?

Comment (dmatej) [reply modules/http-servlet/src/main/java/org/glassfish/grizzly/servlet/SessionCookieConfig.java:33]: Good question, but I don't know after 3 months, perhaps I should make a comment in the commit. It seems to me as a mistake now.

**Pull request merged**